gtsam/hybrid/HybridNonlinearFactorGraph.cpp

```diff
@@ -237,12 +236,17 @@ HybridNonlinearFactorGraph HybridNonlinearFactorGraph::restrict(
     if (auto hf = dynamic_pointer_cast<HybridFactor>(f)) {
       result.push_back(hf->restrict(discreteValues));
     } else if (auto df = dynamic_pointer_cast<DiscreteFactor>(f)) {
-      result.push_back(df->restrict(discreteValues));
+      auto restricted_df = df->restrict(discreteValues);
+      // In the case where all the discrete values have been selected,
```

Comment:
I don't fully understand this comment. No need to reply just see if you can make comment more clear?

Reply:
Done